apps/qubit/modules/user/actions/clipboardSaveAction.class.php

```diff
@@ -100,13 +99,13 @@ private function saveClipboard($allSlugs, $password)
     $save->save();
 
     // Store clipboard items in database
-    foreach($allSlugs as $className => $slugs)
+    foreach($allSlugs as $type => $slugs)
     {
       foreach($slugs as $slug)
       {
         $item = new QubitClipboardSaveItem;
         $item->saveId = $save->id;
-        $item->itemClassName = $className;
+        $item->itemClassName = 'Qubit'.ucfirst($type);
         $item->slug = $slug;
```

Comment:
Yeah, great point about validating. I think that should be enough for now for a couple of reasons:

- If we go with a foreign key, I'd prefer to use the `slug.id` and that will require a few more changes.
- The clipboard could get easily out of sync in the previous and this implementation when resources are edited/deleted. Ideally, we'd have a process to make reflect those changes in the saves, but also in the current clipboards. This can get complicated and after all, in the processes that actually load the resources (view and export), the missing slugs will be filtered in the query.

Reply:
Added validation on clipboard save but the out of sync issue is a bigger problem. I'll add a note in the ticket about it to get feedback from the analysts too.

Reply:
https://projects.artefactual.com/issues/11556#note-8